cluster-autoscaler/simulator/scheduling/hinting_simulator.go

```diff
@@ -62,20 +57,20 @@ func (s *HintingSimulator) TrySchedulePods(clusterSnapshot clustersnapshot.Clust
 	loggingQuota := klogx.PodsLoggingQuota()
 	for _, pod := range pods {
 		klogx.V(5).UpTo(loggingQuota).Infof("Looking for place for %s/%s", pod.Namespace, pod.Name)
-		nodeName, err := s.findNodeWithHints(clusterSnapshot, pod, isNodeAcceptable)
+		nodeName, err := s.tryScheduleUsingHints(clusterSnapshot, pod, isNodeAcceptable)
 		if err != nil {
 			return nil, 0, err
 		}
 
 		if nodeName == "" {
-			nodeName = s.findNode(similarPods, clusterSnapshot, pod, loggingQuota, isNodeAcceptable)
+			nodeName, err = s.trySchedule(similarPods, clusterSnapshot, pod, loggingQuota, isNodeAcceptable)
```

Comment:
I'm curious what the upstream ramifications of this new error condition have. Would it ever make sense to have `breakOnFailure` set to false here?

Reply:
My intention was for `trySchedule` and `tryScheduleUsingHints` to only fail on the "unexpected" errors (to cover the `ForceAddPod` error condition that was previously below, and now is a part of `SchedulePod`, which is called in `trySchedule` and `tryScheduleUsingHints`), and return `"", nil` if the predicates don't pass. So if I got that right `breakOnFailure`, and `TrySchedulePods` in general should work the same as before (modulo the "truly unexpected" errors from `SchedulerPluginRunner` that I mention in the other comment).

@BigDarkClown Could you verify this one as well?

Reply:
As I read it, the difference is that instead of the previous flow (find node -> add pod) we moved to (find node and schedule pod) in a single call. I think this is semantically the same, the only difference is that the `TrySchedulePods` will exit earlier in its body than before , but still on the same errors (scheduling). This looks good to me.

Reply:
I added comments to the `trySchedule` methods and refactored `tryScheduleUsingHints` a bit to make things clearer here.

Reply:
comments help here, I don't love the density of stuff in these flows but that's all inherited, the changes here are appropriately evolutionary, thx!